app/pages/share/ShareInvite.js

```diff
@@ -0,0 +1,410 @@
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { translate } from 'react-i18next';
+import { push } from 'connected-react-router';
+import styled from 'styled-components';
+import { Box, Flex, Text } from 'rebass/styled-components';
+import { Radio, Label } from '@rebass/forms';
+import baseTheme from '../../themes/baseTheme';
+import { useFormik } from 'formik';
+import find from 'lodash/find';
+import get from 'lodash/get';
+import * as yup from 'yup';
+import InputMask from 'react-input-mask';
+import cx from 'classnames';
+
+import { Body1, Caption, Title } from '../../components/elements/FontStyles';
+
+import { useToasts } from '../../providers/ToastProvider';
+import Button from '../../components/elements/Button';
+import TextInput from '../../components/elements/TextInput';
+import Checkbox from '../../components/elements/Checkbox';
+import * as actions from '../../redux/actions';
+import { getCommonFormikFieldProps, fieldsAreValid } from '../../core/forms';
+import { useIsFirstRender } from '../../core/hooks';
+
+const StyledRadio = styled(Radio)`
+  color: ${baseTheme.colors.border.default};
+  width: 1.5em;
+  height: 1.5em;
+  margin-right: 0.5em;
+
+  cursor: pointer;
+  transition: ${baseTheme.transitions.easeOut};
+  position: relative;
+
+  &.checked {
+    color: ${baseTheme.colors.purpleMedium};
+  }
+`;
+
+const ShareInvite = (props) => {
+  const { t, api, trackMetric } = props;
+  const isFirstRender = useIsFirstRender();
+  const dispatch = useDispatch();
+  const { set: setToast } = useToasts();
+  const clinics = useSelector((state) => state.blip.clinics);
+  const loggedInUserId = useSelector((state) => state.blip.loggedInUserId);
+  const { sendingClinicInvite, fetchingClinic, sendingInvite } = useSelector(
+    (state) => state.blip.working
+  );
+  const [clinic, setClinic] = useState(null);
+
+  const shareCodeRegex =
+    /^[ABCDEFGHJKLMNPQRSTUVWXYZ23456789]{4}-[ABCDEFGHJKLMNPQRSTUVWXYZ23456789]{4}-[ABCDEFGHJKLMNPQRSTUVWXYZ23456789]{4}$/;
+
+  const validationSchema = yup.object().shape({
+    type: yup.string().oneOf(['member', 'clinic']).required(),
+    shareCode: yup.string().when('type', {
+      is: 'clinic',
+      then: (schema) =>
+        schema
+          .matches(shareCodeRegex, t('Please enter a valid share code'))
+          .required(t('Please enter a share code')),
+      otherwise: (schema) => schema.notRequired(),
+    }),
+    email: yup.string().when('type', {
+      is: 'member',
+      then: (schema) =>
+        schema
+          .email(t('Please enter a valid email address'))
+          .required(t('Email address is required')),
+      otherwise: (schema) => schema.notRequired(),
+    }),
+    uploadPermission: yup.boolean(),
+  });
+
+  const formikContext = useFormik({
+    initialValues: {
+      type: 'clinic',
+      email: '',
+      shareCode: '',
+      uploadPermission: true,
+    },
+    onSubmit: (values, { setSubmitting }) => {
```

Comment:
```suggestion
    onSubmit: (values) => {
```
`setSubmitting` is unused - can be removed

Reply:
Ah yeah, good call.